packit/cli/propose_downstream.py

```diff
@@ -109,14 +109,26 @@ def downstream(
     VERSION argument is optional, the latest upstream version
     will be used by default
     """
+
     api = get_packit_api(
         config=config, dist_git_path=dist_git_path, local_project=path_or_url
     )
+
     default_dg_branch = api.dg.local_project.git_project.default_branch
```

Comment:
What about doing this call when we know we will use it?

(This will ease the tests as well.)

Reply:
I do apologize, but I don't understand what exactly you mean.

Do you mean to create a function that would encapsulate  determination of dist_git_branch? It is probably a good idea to create such function (eg. `get_dg_branch()`), it would be much easier to test its functionality (in comparsion with whole `propose_downstream()`)

Reply:
That's not what I meant but definitely a good idea..;)

The `api.dg.local_project.git_project.default_branch` use remote service (Pagure) to get that information.

But when looking on the following code, it's not used when we have branches set as CLI argument or in the config:

```python
dg_branches = (
        cmdline_dg_branches or config_dg_branches or default_dg_branch.split(",")
    )
```

My point was that if we can call remote service only when we know that `cmdline_dg_branches` and `config_dg_branches` is not set and we need to do that remote call.

Reply:
> My point was that if we can call remote service only when we know that `cmdline_dg_branches` and `config_dg_branches` is not set and we need to do that remote call.

We do need to call the remote service, because we need to know default git branch in all cases. It is used by `packit.config.aliases.get_branches()` to translate 'rawhide' to corresponding branch name.

Reply:
Ok, good to know.